Fork_Maintenance.md

```diff
@@ -0,0 +1,263 @@
+
+# Create a branch for your feature/fix/patch
+git checkout -b feature/my-new-widget
+# Or for an external PR you want to test/integrate temporarily
+git checkout -b feat/try-upstream-pr-456
+```
+
+**b. Develop and Commit:**
+
+*   **Custom Features/Fixes:** Make your code changes on the feature branch. Commit frequently with clear messages.
+*   **External Patches/PRs:** Apply the patch or cherry-pick commits from the external PR onto your feature branch.
+    ```bash
+    # Example: Cherry-pick commits for PR #456
+    git fetch upstream pull/456/head:upstream-pr-456
+    # Identify commits from the PR branch (e.g., using git log upstream-pr-456)
+    git cherry-pick <commit-sha-1> <commit-sha-2> ...
+    ```
+
+**c. Keep Feature Branch Updated (Optional but Recommended):**
+
+If `release/stable-with-patches` is updated (e.g., after a rebase onto `main`) while you are still working on your feature, rebase your feature branch onto the latest `release/stable-with-patches` to minimize future merge conflicts.
+
+```bash
+git checkout feature/my-new-widget
+git pull origin release/stable-with-patches --rebase # Or git rebase origin/release/stable-with-patches
+# Resolve conflicts if any
+git push origin feature/my-new-widget --force-with-lease
+```
+
+**d. Create Pull Request (PR):**
+
+Once your feature/fix is complete and tested locally, push the feature branch to your origin and create a Pull Request (PR) targeting the `release/stable-with-patches` branch.
+
+```bash
+git push origin feature/my-new-widget
+# Go to your Git hosting platform (e.g., GitHub) and create a PR
+# Base branch: release/stable-with-patches
+# Compare branch: feature/my-new-widget
+```
+
+**e. Code Review and Merge:**
+
+Follow your team's code review process. Once the PR is approved, merge it into `release/stable-with-patches` (typically using the merge button on the Git hosting platform, often configured to use squash or merge commits).
+
+**f. Automatic Build and Deployment:**
+
+The merge into `release/stable-with-patches` should automatically trigger the CI/CD pipeline (Section 2) to build, tag, and push new container images.
+
+**g. Removing Patches:**
+
+*   **External PR Merged Upstream:** If an external PR you integrated via a feature branch is merged into `upstream/main`, the changes will eventually be incorporated into `release/stable-with-patches` when it's rebased onto `main` (Section 3d). You can simply delete your temporary feature branch (`feature/try-upstream-pr-456`). The rebase of `release/stable-with-patches` might require conflict resolution if your temporary integration conflicts with the official merge.
+*   **Custom Feature No Longer Needed:** If a custom feature merged into `release/stable-with-patches` needs to be reverted, use `git revert` on the `release/stable-with-patches` branch to undo the merge commit (or the specific commits if squashed). This creates a new commit that undoes the changes.
+
+## 5. Workflow Summary
```

Comment:
Not sure this section is meaningful to discuss, most of the above questions will have answers that change this significantly.